### PR TITLE
Split OrleansSQLUtils into separate packages

### PR DESCRIPTION
**Pull request by attilah**
Split OrleansSQLUtils into separate packages:
- Orleans.Clustering.AdoNet
- Orleans.Persistence.AdoNet
- Orleans.Reminders.AdoNet
- Orleans.Statistics.AdoNet

Keep OrleansSQLUtils as Meta Package

Breaking change: Internalize shared types (like for Azure and AWS)

### CODE REVIEW

**Comment by attilah** [comment]
@veikkoeeva please review it.

File: src/AdoNet/Orleans.Persistence.AdoNet/Orleans.Persistence.AdoNet.csproj [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <PackageId>Microsoft.Orleans.Persistence.AdoNet</PackageId>
+    <Title>Microsoft Orleans Persistence AdoNet Providers</Title>
+    <Description>Microsoft Orleans persistence providers for AdoNet Providers</Description>
```
Comment (veikkoeeva) [inline src/AdoNet/Orleans.Persistence.AdoNet/Orleans.Persistence.AdoNet.csproj]: Should these "AdoNet"s be _ADO.NET_ here? Are these used as search words in Nuget? ADO.NET is the official term, I think.

File: src/AdoNet/Orleans.Reminders.AdoNet/Orleans.Reminders.AdoNet.csproj [hunk 2]
```diff
@@ -0,0 +1,86 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <PackageId>Microsoft.Orleans.Reminders.AdoNet</PackageId>
+    <Title>Microsoft Orleans Reminders AdoNet Providers</Title>
+    <Description>Microsoft Orleans reminders provider for AdoNet Providers</Description>
```
Comment (veikkoeeva) [inline src/AdoNet/Orleans.Reminders.AdoNet/Orleans.Reminders.AdoNet.csproj]: As previously mentioned about "AdoNet" -> "ADO.NET", here too. I suppose in the other packages also.

File: src/AdoNet/Shared/Storage/DbExtensions.cs [hunk 3]
```diff
@@ -7,12 +7,24 @@
...
+namespace Orleans.Tests.SqlUtils
+#else
+// No default namespace intentionally to cause compile errors if something is not defined
+#endif
 {
     /// <summary>
     /// Contains some convenience methods to use in conjunction with <see cref="IRelationalStorage">IRelationalStorage</see> and <see cref="RelationalStorage">GenericRelationalStorage</see>.
     /// </summary>
-    public static class DbExtensions
+    internal static class DbExtensions
```
Comment (veikkoeeva) [inline src/AdoNet/Shared/Storage/DbExtensions.cs]: Just a note about this and other changes to visibility, these can be used as a micro-ORM like Dapper, for instance. If someone has done that, this is a breaking change. If someone wish to do so, it's not possible anymore after this change.

File: src/AdoNet/OrleansSQLUtils/OrleansSQLUtils.csproj [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <PackageId>Microsoft.Orleans.OrleansSQLUtils</PackageId>
+    <Title>Microsoft Orleans SQL Utilities</Title>
+    <Description>Library of utility types for relational storage of Microsoft Orleans.</Description>
```
Comment (veikkoeeva) [inline src/AdoNet/OrleansSQLUtils/OrleansSQLUtils.csproj]: The other descriptions do not have '.' here, should this one? I don't know how it has been.

File: src/AdoNet/Shared/Storage/DbConnectionFactory.cs [hunk 5]
```diff
@@ -5,7 +5,19 @@
 using System.Linq;
 using System.Reflection;
 
-namespace Orleans.SqlUtils
```
Comment (veikkoeeva) [inline src/AdoNet/Shared/Storage/DbConnectionFactory.cs]: Could there be one main package that have these common parts to every other functionality which the persistence, reminders, statistics and membership implementations then reference? This way, it seems by reading this part, one could avoid some of the conditional namespaces.

File: src/AdoNet/Shared/Storage/DbExtensions.cs (hunk 3 above)
Comment (attilah) [reply src/AdoNet/Shared/Storage/DbExtensions.cs]: Yes we're aware of that, but this is the solution we chose to share files between pieces.

File: src/AdoNet/Shared/Storage/DbConnectionFactory.cs (hunk 5 above)
Comment (attilah) [reply src/AdoNet/Shared/Storage/DbConnectionFactory.cs]: That was one another idea we had, having a "Core" package, but decided to share files instead.

**Comment by veikkoeeva** [comment]
I don't think I have further to add. I would be intriqued to know about why sharing the files and change namespace instead of defining a core package since it seems a bit more complicated. Maybe that is also the reason for the change of some of the parts to internal in addition to, perhaps, reducing API surface.

**Pull request merged**